pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -59,6 +59,24 @@ func (m MemPerCPU) String() string {
 	return "unknown"
 }
 
+// LocalSSDSetting controls whether test cluster nodes use an instance-local SSD
+// as storage.
+type LocalSSDSetting int
+
+const (
+	// LocalSSDDefault is the default mode, when the test does not have any
+	// preference. A local SSD may or may not be used, depending on --local-ssd
+	// flag and machine type.
```

Comment:
Sigh, I wish this was simpler. "Default" could be metamorphic but currently it's defined by `--local-ssd`. The interplay between all of these components (and which one prevails) is pretty arbitrary to me (for example, IMO typically command line flags exist to overrite settings in the code, but in this case, the flag itself is overwritten).

I'd be pretty confused, as a test author. It's hard to reason about when a test is or is not going to use local SSD.

In any case, nothing to change here as this is all existing behaviour; just a rant.

Reply:
And then there is `--roachprod-local-ssd` which actually does override. I agree, it's a mess. I'd be happy to implement if we decide on a desirable and coherent behavior.

We *could* make the `Default` metamorphic unless the `--local-ssd` (or `--local-ssd=false`) flag is passed. But yes I'd want to separate those changes from this refactoring PR.